### PR TITLE
rustdoc: fix position of `default` in method rendering

### PR DESCRIPTION
**Pull request by wackbyte**
With the following code:
```rs
#![feature(specialization)]

pub trait A {
    unsafe fn a();
}

impl A for () {
    default unsafe fn a() {}
}
```
rustdoc would render the `impl` of `a` as
```rs
unsafe default fn a()
```
which is inconsistent with the actual position of `default`.
This PR fixes this issue.

### CODE REVIEW

**Review by fmease — CHANGES_REQUESTED** [review]
Thanks! Could you add a test for this though?

File: tests/rustdoc/default-trait-method.rs [hunk 1]
```diff
@@ -1,26 +1,51 @@
...
+    // @has - '//*[@id="method.foo"]' 'default fn foo()'
     default fn foo() {}
+
+    // @has - '//*[@id="method.bar"]' 'fn bar()'
+    // @!has - '//*[@id="method.bar"]' 'default fn bar()'
     fn bar() {}
+
+    // @has - '//*[@id="method.baz"]' 'unsafe fn baz()'
+    // @has - '//*[@id="method.baz"]' 'default unsafe fn baz()'
+    // @!has - '//*[@id="method.baz"]' 'unsafe default fn baz()'
```
Comment (fmease) [inline tests/rustdoc/default-trait-method.rs:41]:
```suggestion
    // @has - '//*[@id="method.baz"]' 'default unsafe fn baz()'
```

The first check is definitely redundant I'd say. I'm not sure how useful the third one is.

**Review by fmease — APPROVED** [review]
One nit, then it's good from my side at least. Thanks for adding the test cases.

**Review by fmease — APPROVED** [review]

**Review by GuillaumeGomez — APPROVED** [review]

**Pull request merged**